Source/WebKit/WebKitSwift/StageMode/WKStageMode.swift

```diff
@@ -0,0 +1,110 @@
+
+internal import CoreRE
+import Combine
+import Foundation
+import RealityKit
+import WebKitSwift
+import os
+import simd
+internal import Spatial
+@_spi(Private) @_spi(RealityKit) @_spi(CoreREAdditions) import RealityFoundation
+
+fileprivate extension Logger {
+    static let objectManipulation = Logger(subsystem: "com.apple.WebKit", category: "StageModeInteraction")
+}
+
+/// A driver that maps all gesture updates to the specific transform we want for the specified StageMode behavior
+@MainActor
+@objc(WKStageModeInteractionDriver)
+public final class WKStageModeInteractionDriver: NSObject {
+    private var stageModeOperation: WKStageModeOperation = .none
+    let interactionTarget: Entity
+    
+    @objc override init() {
+        self.interactionTarget = Entity()
+    }
+    
+    init(_ target: Entity) {
+        Logger.objectManipulation.debug("WKOMInteraction: WKStageModeInteractionDriver created driver for target \(target.name)")
+        self.interactionTarget = target
+    }
+    
+    // MARK: ObjC Exposed API
+    @objc(initWithInteractionTarget:model:container:)
+    convenience init(with entity: REEntityRef, model: REEntityRef, container: REEntityRef) {
+        let interactionContainer = Entity.__fromCore(__EntityRef.__fromCore(entity))
+        let modelEntity = Entity.__fromCore(__EntityRef.__fromCore(model))
+        let containerEntity = Entity.__fromCore(__EntityRef.__fromCore(container))
+        
+        interactionContainer.setPosition(modelEntity.visualBounds(relativeTo: nil).center, relativeTo: nil)
+        modelEntity.setParent(interactionContainer, preservingWorldTransform: true)
+        interactionContainer.setParent(containerEntity, preservingWorldTransform: true)
+        
+        self.init(interactionContainer)
+    }
+    
+    @objc(initWithInteractionTarget:wkModel:container:)
+    convenience init(with entity: REEntityRef, wkModel: WKSRKEntity?, container: REEntityRef) {
+        guard let wkModel else {
+            self.init()
+            return
+        }
+        
+        let interactionContainer = Entity.__fromCore(__EntityRef.__fromCore(entity))
+        let modelEntity = wkModel.entity
+        let containerEntity = Entity.__fromCore(__EntityRef.__fromCore(container))
+        
+        interactionContainer.setPosition(modelEntity.visualBounds(relativeTo: nil).center, relativeTo: nil)
+        modelEntity.setParent(interactionContainer, preservingWorldTransform: true)
+        interactionContainer.setParent(containerEntity, preservingWorldTransform: true)
+        
+        self.init(interactionContainer)
+    }
+    
+    @objc(interactionDidBegin:)
+    func interactionDidBegin(_ transform: simd_float4x4) {
+        // TODO: rdar://141251753
+    }
+    
+    @objc(interactionDidUpdate:)
+    func interactionDidUpdate(_ transform: simd_float4x4) {
+        // TODO: rdar://141251753
+    }
+    
+    @objc(interactionDidEnd)
+    func interactionDidEnd() {
+        // TODO: rdar://141251753
```

Comment:
nit - WebKit uses FIXME, rather than TODO. If possible, these should also reference bugzilla bugs instead of radars.